ivadomed/loader/segmentation_pair.py

```diff
@@ -315,12 +315,12 @@ def convert_file_to_nifti(self, filename: str, extension: str, is_gt: bool = Fal
         """
         # For '.png', '.tif', '.tiff', '.jpg' and 'jpeg' extentions
         # Read image as 8 bit grayscale in numpy array (behavior TBD in ivadomed for RGB, RBGA or higher bit depth)
-        if "tif" in extension:
-            img = np.expand_dims(imageio.v2.imread(filename, format='tiff-pil'), axis=-1).astype(np.uint8)
-            if len(img.shape) > 3:
-                img = np.expand_dims(imageio.v2.imread(filename, format='tiff-pil', as_gray=True), axis=-1).astype(np.uint8)
-        else:
-            img = np.expand_dims(imageio.v2.imread(filename, as_gray=True), axis=-1).astype(np.uint8)
+        _img = imageio.v3.imread(filename)
+        
+        if len(_img.shape) < 3:
```

Comment:
```suggestion
        if len(_img.shape) >= 3:
            if extension == '.tif' or extension == '.tiff':
                # following line does not work !!!
                _img = imageio.v3.imread(filename, plugin='pillow', extension='.tif', mode='L')
            else:
                _img = imageio.v3.imread(filename, plugin='pillow', mode='L')
        _img = np.expand_dims(_img, axis=-1)
```

After extensive testing, I figured the only way to get back the RGB/RGBA compat was to use the previous logic. Now, it works for PNG images, but I wasn't able to make it work for TIFF images. I think the appropriate backend for TIFF would be `plugin='tifffile'` instead of `plugin='pillow'`, but the tifffile API is obscure to me...